### ISSUE TITLE
ALB is not created with correct IP address type (dualstack, ipv4)

### ISSUE THREAD

**Issue by bradreardon** [kubernetes-sigs/aws-alb-ingress-controller#395]
Was tinkering around with enabling IPv6 on our ALBs and noticed that setting `alb.ingress.kubernetes.io/ip-address-type: dualstack` in the annotations for my ingress resource did not result in the ALB being created with dualstack networking on AWS, but always ipv4.

Relevant annotations from my ingress resource:
```
  annotations:
    kubernetes.io/ingress.class: "alb"
    alb.ingress.kubernetes.io/listen-ports: '[{"HTTP": 80,"HTTPS": 443}]'
    alb.ingress.kubernetes.io/scheme: internet-facing
    alb.ingress.kubernetes.io/subnets: subnet-xxx, subnet-xxx
    alb.ingress.kubernetes.io/security-groups: sg-xxx
    alb.ingress.kubernetes.io/certificate-arn: {{ .Values.ingress.certificateArn }}
    alb.ingress.kubernetes.io/healthcheck-path: /xxx
    alb.ingress.kubernetes.io/ip-address-type: dualstack
```

**Comment by bigkraig** [comment]
@bradreardon which image tag are you using?

**Comment by bradreardon** [comment]
@bigkraig looks like `1.0-alpha.3` which is the tag included in the example manifests.

Any newer, production-tested tags I could be running off of?

**Comment by bigkraig** [comment]
@bradreardon there was a regression and a fix has been submitted in #385, i'm waiting on some testing being done by @khrisrichardson before I merge it in.

**Comment by bigkraig** [comment]
@bradreardon Did you add the annotation to an existing ingress or was that in a new ingress?

**Comment by zioalex** [comment]
I ran in the same problem.
Everything is working properly expect the IPv6 R53 record. Also the ALB has been created without any AAAA record.

**Comment by lareeth** [comment]
@zioalex External DNS is managed separately now, I'm working on getting IPv6 enabled, you can test a preview here: https://github.com/kubernetes-incubator/external-dns/pull/576

**Comment by zioalex** [comment]
Thanks @lareeth will do.

**Comment by zioalex** [comment]
@lareeth  how do I test it? I tried to add the annotation
```external-dns.alpha.kubernetes.io/ipv6: true```

To the external-dns Deployment but it is not recognized. May be because the external-dns code was not yet merged?

**Comment by lareeth** [comment]
@zioalex its still a work in progress, that feature is still being worked on. Lets move the discussion to external-dns and close this issue here.

**Comment by bigkraig** [comment]
Is there anything that needs to be addressed in the ingress controller? I do not have a network to do a full test against, but when changing to dualstack I can see mine failing to reconcile the state due to a lack of a ipv6 subnet.

**Comment by lareeth** [comment]
@bigkraig You will need subnets with ipv6 enabled. Do we need to add better error catching to make this clearer in the logs?

Do we also need to update the documentation to make this clearer as well?

**Comment by bigkraig** [comment]
Its clear, AWS exposes the message that I need ipv6. I'm not sure what needs to be updated/changed at this point. There shouldn't be any DNS records managed.

**Comment by zioalex** [comment]
Aldo the routing tables need to be updated or the traffic will be not forwarded.

**Comment by zioalex** [comment]
So just to recap my experience. The VPC should be ready for IPv6 support otherwise all this will not work.

The annotation:
```alb.ingress.kubernetes.io/ip-address-type: dualstack```
Is not working and also not throwing out any error.

The DNS AAAA records will be part of https://github.com/kubernetes-incubator/external-dns/pull/576

Requirements:
* IPv6 VPC

**Comment by rvaralda** [comment]
+1

**Comment by zioalex** [comment]
@lareeth anything I can help here to merge it?

**Comment by bradreardon** [comment]
Looks like this was fixed in #418 but it never made it back to this thread.

/close
